File: 1]. DSA/1]. Data Structures/01]. Array/C++/_032)_Maximum_length_Even_Odd_Subarray.cpp
#include<iostream>
using namespace std;

int maxEvenOdd(int arr[], int n)
{
    int res = 1, curr = 1;
    for(int i=1; i<n; i++)
    {
        if((arr[i]%2 == 0 && arr[i-1]%2 != 0) ||
               (arr[i]%2 != 0 && arr[i-1]%2 == 0))
        {
            curr++;
            res = max(res, curr);
        }
        else
            curr = 1;
    }
    return res;
}

int main()
{
    int arr[] = {10, 12, 14, 7, 8}, n=5;
    cout<<"Maximum Length of Even-Odd = "<<maxEvenOdd(arr, n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_032)_Maximum_length_Even_Odd_Subarray.dart
import 'dart:math';

int maxEvenOdd(List<int> arr) {
  int res = 1;
  int curr = 1;

  for (int i = 1; i < arr.length; i++) {
    if ((arr[i] % 2 == 0 && arr[i - 1] % 2 != 0) ||
        (arr[i] % 2 != 0 && arr[i - 1] % 2 == 0)) {
      curr++;
      res = max(res, curr);
    } else {
      curr = 1;
    }
  }

  return res;
}

void main() {
  var arr = [10, 12, 14, 7, 8];
  print('Maximum length of Even Odd Subarray: ${maxEvenOdd(arr)}');
}
